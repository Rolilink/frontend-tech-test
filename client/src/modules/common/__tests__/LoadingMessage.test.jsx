import React from 'react';
import { shallow } from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import LoadingMessage from '../LoadingMessage';

expect.addSnapshotSerializer(serializer);
const renderMessage = () => <div className="message" />;

describe('Common.LoadingMessage', () => {
  it('should render children when isLoading prop is false', () => {
    const isLoading = false;
    const children = <div className="children" />;

    const loadingMessage = shallow(
      <LoadingMessage isLoading={isLoading}>
        {children}
      </LoadingMessage>,
    );

    expect(loadingMessage).toMatchSnapshot();
  });

  it('should render a message when isLoading prop is true', () => {
    const isLoading = true;
    const children = <div className="children" />;

    const loadingMessage = shallow(
      <LoadingMessage isLoading={isLoading} renderMessage={renderMessage}>
        {children}
      </LoadingMessage>,
    );

    expect(loadingMessage).toMatchSnapshot();
  });
});
