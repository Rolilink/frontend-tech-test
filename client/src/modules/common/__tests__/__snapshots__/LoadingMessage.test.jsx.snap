// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common.LoadingMessage should render a message when isLoading prop is true 1`] = `
<div
  className="LoadingMessage "
>
  <div
    className="LoadingMessage-Spinner"
  >
    <Spinner />
  </div>
  <div
    className="LoadingMessage-Message"
  >
    <div
      className="message"
    />
  </div>
</div>
`;

exports[`Common.LoadingMessage should render children when isLoading prop is false 1`] = `
<div
  className="children"
/>
`;
